kubewatcher/kubewatcher.go

```diff
@@ -258,50 +257,74 @@ func getResourceVersion(obj runtime.Object) (string, error) {
 func (ws *watchSubscription) eventDispatchLoop() {
 	log.Println("Listening to watch ", ws.Watch.Metadata.Name)
 	for {
-		for {
-			ev, more := <-ws.kubeWatch.ResultChan()
-			if !more {
+		// check watchSubscription is stopped or not before waiting for event
+		// comes from the kubeWatch.ResultChan(). This fix the edge case that
+		// new kubewatch is created in the restartWatch() while the old kubewatch
+		// is being used in watchSubscription.stop().
+		if ws.isStopped() {
+			ws.stop()
```

Comment:
I don't understand this either. If `ws.isStopped()` is true why do you need to call `ws.stop()`?